### ISSUE TITLE
Can't run commands because error: JSON-RPC error: transport error: unexpected HTTP code: 404

### ISSUE THREAD

**Issue by simongltr** [ordinals/ord#2336]
My ord instance is synced but I can't run commands.

The error is always:
```
ubuntu@ip-10-10-0-71:~$ RUST_LOG=info ord wallet create
[2023-08-18T14:42:45Z INFO  ord::options] Connecting to Bitcoin Core at 127.0.0.1:8332/wallet/ord
error: JSON-RPC error: transport error: unexpected HTTP code: 404
```

Same for `list`, same for `server` ...

I have Bitcoin Core version v24.0.1 and Ord v0.8.2, my `bitcoin.conf` file is:
```
server=1
daemon=1
txindex=1
```

I have never created the "ord" wallet because I can't, and I see that it tries to connect to `127.0.0.1:8332/wallet/ord` that may not exist for this reason ?

I only need to fetch ranges from the `/output` endpoint of the json api.

Thank you,
Simon

**Comment by cypherpork** [comment]
@simongltr, I am assuming your `bitcoind` is running on the same host and when it is running you see a `.cookie` file in your `~/.bitcoin` directory. If so:
1. Choose username and password (un pw) to rpc-connect to your bitcoin node.
2. Generate an `rpcauth` line for yourself. Use this service, for example: https://jlopp.github.io/bitcoin-core-rpc-auth-generator/
3. add some lines to your bitcoin.conf; make it look like this:
```
server=1
daemon=1
txindex=1
listen=1
rpcauth=un:39ccf9fe61c90d00a6eab3483532452e$1600e3f6e324adf4f60053b45cbcae092543c032c6a332887b10500a39962399
# ^^ replace the line above with the one you got from step 2
rpcallowip=0.0.0.0/0
[main]
rpcport=8332
rpcbind = 0.0.0.0
bind=0.0.0.0
```
4. restart `bitcoind`
5. Try this: `ord --rpc-url http://un:pw@127.0.0.1:8332 index run`; If it runs, you are on the right track.
6. Let it finish building an index.
7. Try this: `ord --rpc-url http://127.0.0.1:8332 --cookie-file ~/.bitcoin/.cookie index run`- this is the version of command that does not expose un/pw.
8.  If you are successful in reaching item 7, you can start shaping your desired commands like this. For example, to run a local ord explorer server: `ord --rpc-url http://127.0.0.1:8332 --cookie-file ~/.bitcoin/.cookie server --http --http-port 80 --address 0.0.0.0`
9. once you get everything running, tighten down your `rpcallowip rpcbind bind` to the exact IPs you need
10. and try to get rid of default values in bitcoin.config and in ord command line, which I deliberately left in the examples.

**Comment by so7ow** [comment]
Did you build bitcoin from source? That 404 can happen when bitcoin is built without wallet support. Check your debug.log for messages to that effect.

**Comment by simongltr** [comment]
Yes I was building from source but I have issues with the `BerkeleyDB` dependency. I saw [here](https://github.com/bitcoin/bitcoin/blob/master/doc/build-unix.md#ubuntu--debian) that bitcoin core needs `BerkeleyDB version 4.8` to have full wallet support but it's not clear to me how to find it.

I tried to install from `libdb-dev` and `libdb++-dev` packages but it's not version 4.8. I also tried to install from binary because I was thinking maybe the dependency is included but looks like no...

My current situation is that I have no wallet related commands in bitcoin-cli, which I guess is the issue...

**Comment by simongltr** [comment]
I solved the problem, as discussed it was a problem realated to an incorrect compilation of bitcoin core.

I solved it by installing `libdb-dev` and `libdb++-dev` packages (`sudo apt install ...`) and compiling with the correct flag:
```
git clone https://github.com/bitcoin/bitcoin.git
cd bitcoin/
./autogen.sh
./configure --with-incompatible-bdb
make
sudo make install
```

Now I have access to the wallet feature.

My 404 error is gone but I have a 401 error now (authentification issue ?)

Closing this issue.
